miio/integrations/petfeeder/status.py

```diff
@@ -0,0 +1,85 @@
+from datetime import timedelta
+from enum import Enum
+from typing import Any, Dict
+
+from miio import DeviceStatus
+
+
+class FoodLevel(Enum):
+    Normal = 0
+    Low = 1
+    Empty = 2
+
+
+class FoodOutletStatus(Enum):
+    Ok = 0
+    Blocked = 1
+
+
+class FoodStorageLid(Enum):
+    Closed = 0
+    Open = 1
+
+
+class PetFeederStatus(DeviceStatus):
+    """Container for status reports from the Pet Feeder / Smart Pet Food Dispenser."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """
+        Response from pet feeder (mmgg.feeder.petfeeder):
+
+        {
+            "food_level": 0,
+            "feed_plan": 1,
+            "food_outlet": 0,
+            "undeclared": 0,
+            "clean_days_left": 11,
+            "food_storage_lid": 0,
+            "desiccant_days_left": 21,
+            "weight_level": 0,
+            "wifi_led": 0,
+            "button_lock": 1,
+            "country_code": 255
+        }
+        """
+        self.data = data
+
+    @property
+    def food_level(self) -> FoodLevel:
+        """Current food level / status."""
+        return FoodLevel(self.data["food_level"])
+
+    @property
+    def feed_plan(self) -> bool:
+        """Automatic feeding status."""
+        return bool(self.data["feed_plan"])
+
+    @property
+    def food_outlet(self) -> FoodOutletStatus:
+        """Food outlet status."""
+        return FoodOutletStatus(self.data["food_outlet"])
+
+    @property
+    def food_storage_lid(self) -> FoodStorageLid:
+        """Food storage lid status."""
+        return FoodStorageLid(self.data["food_storage_lid"])
+
+    @property
+    def clean_days_left(self) -> timedelta:
+        """Number of days until the unit requires cleaning."""
+        return timedelta(days=self.data["clean_days_left"])
+
+    @property
+    def desiccant_days_left(self) -> timedelta:
+        """Number of days until the desiccant disc requires replacing."""
+        return timedelta(days=self.data["desiccant_days_left"])
+
+    @property
+    def wifi_led(self) -> bool:
+        """WiFi status led."""
+        return bool(self.data["wifi_led"])
+
+    @property
+    def button_lock(self) -> bool:
```

Comment:
```suggestion
    def child_lock(self) -> bool:
```
Some other devices (fans especially) have a similar setting and are using this name. Let's use the same one here just to be consistent among devices in case someone wants to create a generic interface to control these devices.

Reply:
It's not a child lock...it's closer to being a "clever pet" lock.

There's a button on the unit you can press to dispense 1 unit of food, handy feature to be honest. 
Until you pet (like mine) works this out and thinks it's a great game to press repeatedly until it's empty 🤦

Reply:
Yes I understand, but the principle is the same. But it's also fine as it is.

Reply:
Only in the sense that it's a button that enables of disabled certain functionality. 
Child lock as a name will break readability imho.